### ISSUE TITLE
InvalidAsn1Error: Expected 0x2: got 0xd

### ISSUE THREAD

**Issue by moritzmoritz** [evs-broadcast/node-emberplus#12]
Hey!
Thanks for adding support for matrices.

I'm trying to read some matrices, but I always get following issue:

```
QualifiedMatrix decode
InvalidAsn1Error: Expected 0x2: got 0xd
```

my stack trace:

```
InvalidAsn1Error: Expected 0x2: got 0xd
    at newInvalidAsn1Error (/Users/*/workspace/*/node_ember/node_modules/asn1/lib/ber/errors.js:7:13)
    at ExtendedReader.Reader._readTag (/Users/*/workspace/*/node_ember/node_modules/asn1/lib/ber/reader.js:331:11)
    at ExtendedReader.Reader.readInt (/Users/*/workspace/*/node_ember/node_modules/asn1/lib/ber/reader.js:206:15)
    at Function.MatrixContents.decode (/Users/*/workspace/*/node_ember/node_modules/emberplus/ember.js:847:41)
    at Function.QualifiedMatrix.decode (/Users/*/workspace/*/node_ember/node_modules/emberplus/ember.js:1210:42)
    at Function.Element.decode (/Users/*/workspace/*/node_ember/node_modules/emberplus/ember.js:438:32)
    at Function.RootElement.decode (/Users/*/workspace/*/node_ember/node_modules/emberplus/ember.js:404:20)
    at Function.Root.decode (/Users/*/workspace/*/node_ember/node_modules/emberplus/ember.js:50:50)
    at S101Codec.self.codec.on (/Users/*/workspace/*/node_ember/node_modules/emberplus/client.js:195:39)
```

Any idea what am I doing wrong?

My code:
```
tree.connect()
.then(() => { 
   return tree.getDirectory();
})
.then((r) => {
   root = r ;
   return tree.expand(r.elements[0]);
})
.then((r) => {
	let matrices = root.elements[0].children[1].children;
	let matrix = matrices[0];
})
.catch((e) => {
   console.log(e.stack);
});
```

**Comment by Kaliph** [comment]
I am experiencing exactly the same issue. Would be great to get a hint about what is going on. Thankyou

**Comment by dufourgilles** [comment]
Could you capture a sniffer trace while retrieving ember tree and attach it here.
Thanks.

**Comment by Kaliph** [comment]
Ok. I have made a Wireshark pcap file. I hope it is what you´re looking for.
[TinyEmberPlus.zip](https://github.com/evs-broadcast/node-emberplus/files/2217130/TinyEmberPlus.zip)

**Comment by dufourgilles** [comment]
I did not see a problem in the trace you provided.
All requests got answered and I could decode the last message properly.
I believe the problem is now fix in latest version. 1.7.12.  Could you give it a try ?

**Comment by Kaliph** [comment]
I tried it but the error is still the same. Here is the TinyEmberPlusRouter-1.6.2 I am using
[TinyEmberPlusRouter-1.6.2.zip](https://github.com/evs-broadcast/node-emberplus/files/2290176/TinyEmberPlusRouter-1.6.2.zip)
And the packaged emberplus 1.7.13
[emberplus-1.7.13.tgz.zip](https://github.com/evs-broadcast/node-emberplus/files/2290177/emberplus-1.7.13.tgz.zip)

**Comment by dufourgilles** [comment]
Your application does not start. I created a windows VM (sorry I'm not a Windows user) and started your App.  It was missing msvcp100.dll. Downloaded the file from the Web. 
Still the App crashes at startup.

**Comment by Kaliph** [comment]
I believe you need to install the correct Microsoft C++ Redistributable Package. Currently I am using Windows 10. I also have a VM with Windows 10 running. You can see me installed C++ Redistributable Versions in my screenshot I have attached. Hope it helps.
![cpp_example](https://user-images.githubusercontent.com/19877914/45376649-a59f5680-b5f8-11e8-989f-bf0234a78499.png)

**Comment by Kaliph** [comment]
Great that you can reproduce the issue now. Looking forward to test your fix. Thankyou

**Comment by dufourgilles** [comment]
Problem fixed.  the parametersLocation of a matrix can be encoded with 2 different formats.  We only had implemented 1 of them.

> d.connect().then(() => d.getDirectory()).then(() => d.expand(d.root.elements[0]));
Promise { <pending> }
> parametersLocation
parametersLocation complete
parametersLocation
parametersLocation complete
parametersLocation
parametersLocation complete
parametersLocation
parametersLocation complete

> d.root.elements[0]
QualifiedNode {
  path: '1',
  contents: NodeContents { isOnline: true, identifier: 'router' },
  children: 
   [ QualifiedNode { path: '1.0', contents: [Object], children: [Object] },
     QualifiedNode { path: '1.1', contents: [Object], children: [Object] },
     QualifiedNode { path: '1.2', contents: [Object], children: [Object] },
     QualifiedNode { path: '1.3', contents: [Object], children: [Object] },
     QualifiedNode { path: '1.4', contents: [Object], children: [Object] } ] }
> d.root.elements[0].children[3]
QualifiedNode {
  path: '1.3',
  contents: 
   NodeContents {
     isOnline: true,
     identifier: 'dynamic',
     description: 'Linear N:N with Dynamic Parameters' },
  children: 
   [ QualifiedNode { path: '1.3.1', contents: [Object], children: [Object] },
     QualifiedMatrix { path: '1.3.2', contents: [Object], connections: [Object] } ] }
> [root@localhost emberTest]#

**Comment by dufourgilles** [comment]
Fixed in 1.7.14.1
